### ISSUE TITLE
Flatpak - failed to create sessions

### ISSUE THREAD

**Issue by Pshemas** [jupyterlab/jupyterlab-desktop#662]
Tried to use flatpak version - sadly it didn't work well on my end. When I try to create new session / notebook I get "failed to create sessions" error.

![image](https://github.com/jupyterlab/jupyterlab-desktop/assets/6144330/d7016834-2842-4945-80f3-95c15423cb26)

Tried reinstalling the python env. via the app menu, but that didn't help.

When checking out the output when starting the flatpaked app via cli I see this:

```
[2:0601/171334.632895:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: Nie ma takiego pliku ani katalogu
[2:0601/171334.632928:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: Nie ma takiego pliku ani katalogu
[41:0601/171336.549351:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
[41:0601/171336.553769:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times!
[41:0601/171337.837378:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 3 times!

```

I'm running it under OpenSUSE TW, Gnome (Wayland session).

**Comment by Pshemas** [comment]
it's happens with rpm as well. Also tried previous (3.6) version and it didn't work. I've attached `main.log` .

When I tried spawning server with:

```
./python -m jupyterlab --no-browser --expose-app-in-browser
```
from the jlab_server/bin folder I got:

```
./python -m jupyterlab --no-browser --expose-app-in-browser
Traceback (most recent call last):
  File "/home/przemek/.config/jupyterlab-desktop/jlab_server/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/przemek/.config/jupyterlab-desktop/jlab_server/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/przemek/.config/jupyterlab-desktop/jlab_server/lib/python3.8/site-packages/jupyterlab/__main__.py", line 6, in <module>
    from jupyterlab.labapp import main
  File "/home/przemek/.config/jupyterlab-desktop/jlab_server/lib/python3.8/site-packages/jupyterlab/labapp.py", line 45, in <module>
    from .extensions import MANAGERS as EXT_MANAGERS
  File "/home/przemek/.config/jupyterlab-desktop/jlab_server/lib/python3.8/site-packages/jupyterlab/extensions/__init__.py", line 12, in <module>
    from .pypi import PyPIExtensionManager
  File "/home/przemek/.config/jupyterlab-desktop/jlab_server/lib/python3.8/site-packages/jupyterlab/extensions/pypi.py", line 23, in <module>
    from async_lru import alru_cache
  File "/home/przemek/.config/jupyterlab-desktop/jlab_server/lib/python3.8/site-packages/async_lru/__init__.py", line 23, in <module>
    from typing_extensions import Self
ImportError: cannot import name 'Self' from 'typing_extensions' (/home/przemek/.local/lib/python3.8/site-packages/typing_extensions.py)
przemek@domekstacja:~/.config/jupyterlab-desktop/jlab_server/bin> ./python3 -m jupyterlab --no-browser --expose-app-in-browser
Traceback (most recent call last):
  File "/home/przemek/.config/jupyterlab-desktop/jlab_server/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/przemek/.config/jupyterlab-desktop/jlab_server/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/przemek/.config/jupyterlab-desktop/jlab_server/lib/python3.8/site-packages/jupyterlab/__main__.py", line 6, in <module>
    from jupyterlab.labapp import main
  File "/home/przemek/.config/jupyterlab-desktop/jlab_server/lib/python3.8/site-packages/jupyterlab/labapp.py", line 45, in <module>
    from .extensions import MANAGERS as EXT_MANAGERS
  File "/home/przemek/.config/jupyterlab-desktop/jlab_server/lib/python3.8/site-packages/jupyterlab/extensions/__init__.py", line 12, in <module>
    from .pypi import PyPIExtensionManager
  File "/home/przemek/.config/jupyterlab-desktop/jlab_server/lib/python3.8/site-packages/jupyterlab/extensions/pypi.py", line 23, in <module>
    from async_lru import alru_cache
  File "/home/przemek/.config/jupyterlab-desktop/jlab_server/lib/python3.8/site-packages/async_lru/__init__.py", line 23, in <module>
    from typing_extensions import Self
ImportError: cannot import name 'Self' from 'typing_extensions' (/home/przemek/.local/lib/python3.8/site-packages/typing_extensions.py)
```

When I did it without ./ (so the installed python was used) it failed... as it launched Python 2.7 that's available with my system. When I typed:

```
python3 -m jupyterlab --no-browser --expose-app-in-browser
```

and Python 3 has been used the server launched. The app didn't found local JupyterLab servers, but when I've pasted url with token it connected correctly.

**Comment by JasonWeill** [comment]
@Pshemas Thanks for opening this issue! Have you followed the steps in the [JupyterLab Desktop troubleshooting guide](https://github.com/jupyterlab/jupyterlab-desktop/blob/master/troubleshoot.md)? It would help if you provided info about the Python 3 versions you've been using, both with Desktop 3.x and 4.x.

**Comment by JasonWeill** [comment]
@Pshemas Any update on the questions above?

**Comment by zyrxvo** [comment]
I'm also unable to get Jupyter Lab Desktop to create a session. I can connect to remote and local sessions that I instantiate, but it always fails to create a self-contained session. 

I tried the simplest, most up-to-date approach I could think of. I'm using a M2 Pro MacBook Pro (2023) running macOS Ventura 13.4. I used Homebrew to upgrade to Python 3.11.4. I then created a virtual python environment and upgraded pip to 23.1.2 before running `pip install --no-cache jupyterlab`. This is my `pip freeze` at this point.

```
anyio==3.7.0
appnope==0.1.3
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
arrow==1.2.3
asttokens==2.2.1
async-lru==2.0.2
attrs==23.1.0
Babel==2.12.1
backcall==0.2.0
beautifulsoup4==4.12.2
bleach==6.0.0
certifi==2023.5.7
cffi==1.15.1
charset-normalizer==3.1.0
comm==0.1.3
debugpy==1.6.7
decorator==5.1.1
defusedxml==0.7.1
executing==1.2.0
fastjsonschema==2.17.1
fqdn==1.5.1
idna==3.4
ipykernel==6.23.2
ipython==8.14.0
isoduration==20.11.0
jedi==0.18.2
Jinja2==3.1.2
json5==0.9.14
jsonpointer==2.3
jsonschema==4.17.3
jupyter-events==0.6.3
jupyter-lsp==2.2.0
jupyter_client==8.2.0
jupyter_core==5.3.1
jupyter_server==2.6.0
jupyter_server_terminals==0.4.4
jupyterlab==4.0.2
jupyterlab-pygments==0.2.2
jupyterlab_server==2.23.0
MarkupSafe==2.1.3
matplotlib-inline==0.1.6
mistune==2.0.5
nbclient==0.8.0
nbconvert==7.5.0
nbformat==5.9.0
nest-asyncio==1.5.6
notebook_shim==0.2.3
overrides==7.3.1
packaging==23.1
pandocfilters==1.5.0
parso==0.8.3
pexpect==4.8.0
pickleshare==0.7.5
platformdirs==3.5.3
prometheus-client==0.17.0
prompt-toolkit==3.0.38
psutil==5.9.5
ptyprocess==0.7.0
pure-eval==0.2.2
pycparser==2.21
Pygments==2.15.1
pyrsistent==0.19.3
python-dateutil==2.8.2
python-json-logger==2.0.7
PyYAML==6.0
pyzmq==25.1.0
requests==2.31.0
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
Send2Trash==1.8.2
six==1.16.0
sniffio==1.3.0
soupsieve==2.4.1
stack-data==0.6.2
terminado==0.17.1
tinycss2==1.2.1
tornado==6.3.2
traitlets==5.9.0
typing_extensions==4.6.3
uri-template==1.2.0
urllib3==2.0.3
wcwidth==0.2.6
webcolors==1.13
webencodings==0.5.1
websocket-client==1.5.3
```

I then navigated the Jupyter Lab Desktop custom python environment to the `venv/bin/python3.11`.

When I launch Jupyter Lab Desktop, the following error appears in the `main.log`.
```
[2023-06-15 17:10:03.942] [error] Error: Command failed: ln -s /Applications/JupyterLab.app/Contents/Resources/app/jlab /usr/local/bin/jlab
ln: /usr/local/bin/jlab: Permission denied

    at checkExecSyncError (node:child_process:911:11)
    at Object.execSync (node:child_process:983:15)
    at Object.execSync (node:electron/js2c/asar_bundle:2:12711)
    at setupJLabCommand (/Applications/JupyterLab.app/Contents/Resources/app.asar/build/out/main/main.js:151:29)
    at /Applications/JupyterLab.app/Contents/Resources/app.asar/build/out/main/main.js:199:9
```
Then when I try and start a new session, the following error appears.
```
[2023-06-15 17:10:35.959] [error] Failed to start server Error: Failed to launch Jupyter Server
    at /Applications/JupyterLab.app/Contents/Resources/app.asar/build/out/main/server.js:260:28
[2023-06-15 17:10:35.960] [error] Failed to start server Error: Failed to launch Jupyter Server
    at /Applications/JupyterLab.app/Contents/Resources/app.asar/build/out/main/server.js:260:28
[2023-06-15 17:10:35.963] [error] (node:12807) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of undefined (reading 'protocol')
    at new LabView (/Applications/JupyterLab.app/Contents/Resources/app.asar/build/out/main/labview/labview.js:45:50)
    at SessionWindow._loadLabView (/Applications/JupyterLab.app/Contents/Resources/app.asar/build/out/main/sessionwindow/sessionwindow.js:274:25)
    at SessionWindow._updateContentView (/Applications/JupyterLab.app/Contents/Resources/app.asar/build/out/main/sessionwindow/sessionwindow.js:680:18)
    at IpcMainImpl.<anonymous> (/Applications/JupyterLab.app/Contents/Resources/app.asar/build/out/main/sessionwindow/sessionwindow.js:421:18)
(Use `JupyterLab --trace-warnings ...` to show where the warning was created)
[2023-06-15 17:10:35.963] [error] (node:12807) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 3)
```

I also tried completely uninstalling and reinstalling Jupyter Lab Desktop. And I've tried giving Jupyter Lab Desktop Full Disk Access and I've also tried downloading and installing the bundled python environment, but I am unable to get Jupyter Lab Desktop to create a session within the app.

Any help on this would be appreciated and I will provide any additional details that I can to help with this matter.

**Comment by mbektas** [comment]
@zyrxvo can you try to launch the JupyterLab web app from your terminal and see if that reproduces / shows more error details.
```
jlab env activate # ( or '/Applications/JupyterLab.app/Contents/MacOS/JupyterLab' instead of jlab)
jupyter lab

# when done
Ctrl + C
exit
```

**Comment by zyrxvo** [comment]
Thank you for the suggestion. 

Firstly, the `Permission denied` error from launching Jupyter Lab Desktop no longer appears in the `main.log` file. I'm now running macOS Ventura 13.5.2 and Jupyter Lab Desktop Version 4.0.5-1.

Additionally, when I run `jlab env activate` in the terminal, still ***no*** errors appear in `main.log` upon launching. After executing `jupyter lab` in the `jlab` environment, the web app launches normally in the browser without any issues. Here is the terminal output from the interaction.

```zsh

zyrxvo@mymac ~ % jlab env activate
Note: This is an experimental feature.
Activating Python environment "/Users/zyrxvo/Library/jupyterlab-desktop/jlab_server"
(base) bash-5.2$ jupyter lab
[I 2023-09-13 10:57:57.952 ServerApp] Package jupyterlab took 0.0000s to import
[I 2023-09-13 10:57:57.960 ServerApp] Package jupyter_lsp took 0.0081s to import
[W 2023-09-13 10:57:57.960 ServerApp] A `_jupyter_server_extension_points` function was not found in jupyter_lsp. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
[I 2023-09-13 10:57:57.963 ServerApp] Package jupyter_server_terminals took 0.0031s to import
[I 2023-09-13 10:57:57.964 ServerApp] Package notebook_shim took 0.0000s to import
[W 2023-09-13 10:57:57.964 ServerApp] A `_jupyter_server_extension_points` function was not found in notebook_shim. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.
[I 2023-09-13 10:57:57.964 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2023-09-13 10:57:57.966 ServerApp] jupyter_server_terminals | extension was successfully linked.
[I 2023-09-13 10:57:57.968 ServerApp] jupyterlab | extension was successfully linked.
[W 2023-09-13 10:57:57.971 ServerApp] Cannot bind to localhost, using 127.0.0.1 as default ip
    [Errno 8] nodename nor servname provided, or not known
[I 2023-09-13 10:57:58.092 ServerApp] notebook_shim | extension was successfully linked.
[I 2023-09-13 10:57:58.112 ServerApp] notebook_shim | extension was successfully loaded.
[I 2023-09-13 10:57:58.113 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2023-09-13 10:57:58.114 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2023-09-13 10:57:58.114 LabApp] JupyterLab extension loaded from /Users/zyrxvo/Library/jupyterlab-desktop/jlab_server/lib/python3.8/site-packages/jupyterlab
[I 2023-09-13 10:57:58.114 LabApp] JupyterLab application directory is /Users/zyrxvo/Library/jupyterlab-desktop/jlab_server/share/jupyter/lab
[I 2023-09-13 10:57:58.115 LabApp] Extension Manager is 'pypi'.
[I 2023-09-13 10:57:58.116 ServerApp] jupyterlab | extension was successfully loaded.
[I 2023-09-13 10:57:58.117 ServerApp] Serving notebooks from local directory: /Users/zyrxvo
[I 2023-09-13 10:57:58.117 ServerApp] Jupyter Server 2.6.0 is running at:
[I 2023-09-13 10:57:58.117 ServerApp] http://127.0.0.1:8888/lab?token=c68f6d7c3b51c8348bcbb0b636570166e556d4bd328f386a
[I 2023-09-13 10:57:58.117 ServerApp]     http://127.0.0.1:8888/lab?token=c68f6d7c3b51c8348bcbb0b636570166e556d4bd328f386a
[I 2023-09-13 10:57:58.117 ServerApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 2023-09-13 10:57:58.121 ServerApp] 
    
    To access the server, open this file in a browser:
        file:///Users/zyrxvo/Library/Jupyter/runtime/jpserver-26152-open.html
    Or copy and paste one of these URLs:
        http://127.0.0.1:8888/lab?token=c68f6d7c3b51c8348bcbb0b636570166e556d4bd328f386a
        http://127.0.0.1:8888/lab?token=c68f6d7c3b51c8348bcbb0b636570166e556d4bd328f386a
[I 2023-09-13 10:57:58.296 ServerApp] Skipped non-installed server(s): bash-language-server, dockerfile-language-server-nodejs, javascript-typescript-langserver, jedi-language-server, julia-language-server, pyright, python-language-server, python-lsp-server, r-languageserver, sql-language-server, texlab, typescript-language-server, unified-language-server, vscode-css-languageserver-bin, vscode-html-languageserver-bin, vscode-json-languageserver-bin, yaml-language-server
[I 2023-09-13 10:58:01.196 LabApp] Build is up to date

```

If I do try and launch a New Session from the Jupyter Lab Desktop GUI I get an essentially identical error (with changes only to the line and node numbers).

```zsh

[2023-09-13 10:25:33.030] [error] Failed to start server Error: Failed to launch Jupyter Server
    at /Applications/JupyterLab.app/Contents/Resources/app.asar/build/out/main/server.js:284:28
[2023-09-13 10:25:33.035] [error] Failed to start server Error: Failed to launch Jupyter Server
    at /Applications/JupyterLab.app/Contents/Resources/app.asar/build/out/main/server.js:284:28
[2023-09-13 10:25:33.046] [error] (node:22362) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of undefined (reading 'protocol')
    at new LabView (/Applications/JupyterLab.app/Contents/Resources/app.asar/build/out/main/labview/labview.js:45:50)
    at SessionWindow._loadLabView (/Applications/JupyterLab.app/Contents/Resources/app.asar/build/out/main/sessionwindow/sessionwindow.js:274:25)
    at SessionWindow._updateContentView (/Applications/JupyterLab.app/Contents/Resources/app.asar/build/out/main/sessionwindow/sessionwindow.js:680:18)
    at IpcMainImpl.<anonymous> (/Applications/JupyterLab.app/Contents/Resources/app.asar/build/out/main/sessionwindow/sessionwindow.js:421:18)
(Use `JupyterLab --trace-warnings ...` to show where the warning was created)
[2023-09-13 10:25:33.047] [error] (node:22362) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 3)

```

My work around has been to simply connect to a Jupyter Lab session that I manually start in the terminal, but I really enjoyed the elegance of having an 'all-in-one' Jupyter application.

**Comment by zyrxvo** [comment]
I may have found the issue! I'll come back and share more after more testing, but I had a seemingly unrelated issue where I was unable to resolve `localhost` in any browser. This was because of changes I had made to `/etc/hosts` but had long since forgotten about. 

In `/etc/hosts` the `127.0.0.1` line was no longer resolving as `localhost`. After making sure that I did have `127.0.0.1 localhost` I no longer have issues that I was seeing before. When I launch a session in JupyterLab Desktop no errors appear in the `main.log` file. New sessions work as expected with both the bundled python environment as well as my own custom virtual environment.
